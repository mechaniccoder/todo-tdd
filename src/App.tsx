import React from "react";
import { Route, Switch, Link } from "react-router-dom";
import "./App.css";
import Home from "./pages/Home/Home";
import SignIn from "./pages/SignIn/SignIn";
import SignUp from "./pages/SignUp/SignUp";

function App() {
  return (
    <>
      <div>
        <Link to="/">Home</Link>
        <Link to="/signup">Sign Up</Link>
        <Link to="/signin">Sign In</Link>
      </div>
      <Switch>
        <Route path="/" exact component={Home} />
        <Route path="/signup" exact component={SignUp} />
        <Route path="/signin" exact component={SignIn} />
      </Switch>
    </>
  );
}

export default App;
