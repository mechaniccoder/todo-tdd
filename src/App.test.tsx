import { getByText, render, screen } from "@testing-library/react";
import React from "react";
import userEvent from "@testing-library/user-event";
import { BrowserRouter } from "react-router-dom";
import App from "./App";

const renderApp = () => {
  return render(
    <BrowserRouter>
      <App />
    </BrowserRouter>
  );
};

describe("Router", () => {
  it("snapshots App", () => {
    const { container } = renderApp();
    expect(container).toMatchSnapshot();
  });

  it("has 3 Links", () => {
    renderApp();
    expect(screen.getByText("Home")).toBeInTheDocument();
    expect(screen.getByText("Sign Up")).toBeInTheDocument();
    expect(screen.getByText("Sign In")).toBeInTheDocument();
  });

  it("navigates to Home when user clicks Link", () => {
    renderApp();
    userEvent.click(screen.getByText("Home"));
    expect(location.pathname).toBe("/");
    expect(screen.getByText(/go to sign up/)).toBeTruthy();
  });

  it("navigates to Sign Up when user clicks Link", () => {
    renderApp();
    userEvent.click(screen.getByText("Sign Up"));
    expect(screen.getByText(/this is sign up page/)).toBeTruthy();
  });

  it("navigates to Sign In when user clicks Link", () => {
    renderApp();
    userEvent.click(screen.getByText("Sign In"));
    expect(location.pathname).toBe("/signin");
  });
});

export {};
