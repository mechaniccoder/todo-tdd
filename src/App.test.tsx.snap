// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router snapshots App 1`] = `
<div>
  <div>
    <a
      href="/"
    >
      Home
    </a>
    <a
      href="/signup"
    >
      Sign Up
    </a>
    <a
      href="/signin"
    >
      Sign In
    </a>
  </div>
  <div>
    <div>
      go to sign up
    </div>
  </div>
</div>
`;
